app/src/org/commcare/android/resource/installers/OfflineUserRestoreAndroidInstaller.java

```diff
@@ -40,10 +40,13 @@ public OfflineUserRestoreAndroidInstaller(String localDestination, String upgrad
 
     @Override
     public boolean initialize(AndroidCommCarePlatform instance, boolean isUpgrade) throws ResourceInitializationException {
+        instance.registerDemoUserRestore(initDemoUserRestore());
+        return true;
+    }
+
+    private OfflineUserRestore initDemoUserRestore() throws ResourceInitializationException {
         try {
-            OfflineUserRestore offlineUserRestore = new OfflineUserRestore(localLocation);
-            instance.registerDemoUserRestore(offlineUserRestore);
-            return true;
+            return new OfflineUserRestore(localLocation);
         } catch (UnfullfilledRequirementsException e) {
             throw new ResourceInitializationException(e.getMessage());
```

Comment:
One more very, very minor point: We should be wrapping these exceptions properly (IE: initializing the throwable source in the new exception) rather than just adding the same string message, otherwise we lose the most relevant part of the stack trace. This is a general habit we should get into with error handling